### ISSUE TITLE
Domain Events - Add Support

### ISSUE THREAD

**Issue by danielmackay** [event-catalog/eventcatalog#545]
## Problem / Context

Currently, EventCatalog is targeted out Micro Services, the Domains they contain, and the Integration Events that flow between different services (at least from my understanding).

Integration Events are those that flow between Services or Systems.

## Suggestion

It would be great to also have support for Domain Events, which are events that are raised and handled WITHIN a Domain.

The following are typically involved:

- Aggregate Root - Raises the Domain Event
- Domain Event - Contains the data for the Event
- Event Handler - Receives notification that the Event has been raised and actions it

**Comment by black-snow** [comment]
How would you distinguish between 'em in an AsyncAPI spec, for example? I feel like this adds confusion.

I wouldn't want to expose my internal sync APIs in, e.g., Backstage either. Isn't the whole point to document your _public_ API?

**Comment by danielmackay** [comment]
> How would you distinguish between 'em in an AsyncAPI spec, for example? I feel like this adds confusion.
> 
> I wouldn't want to expose my internal sync APIs in, e.g., Backstage either. Isn't the whole point to document your _public_ API?

You are correct, Domain Events are not public, but private within an application or module.  Therefore, it most likely doesn't make sense in an AsyncAPI spec.

However, I believe there is still value in documenting internal Domain Events to help better understand a system.

**Comment by boyney123** [comment]
This how now been added https://www.eventcatalog.dev/blog/domain-level-messages
